sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Testing/ConnectionStringTokenParser.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.RegularExpressions;
+
+namespace Azure.Messaging.EventHubs.Tests
+{
+    /// <summary>
+    ///   General purpose parser that allows to retrieve a token in connection string.
+    /// </summary>
+    ///
+    internal static class ConnectionStringTokenParser
+    {
+        /// <summary>It matches a pair "key=value" in a connection string</summary>
+        private static readonly Regex tokenValueRegex = new Regex("(?<tokenkey>[^;]+)=(?<tokenvalue>[^;]+)(;|$)", RegexOptions.IgnoreCase | RegexOptions.Compiled);
+
+        /// <summary>It matches the name of an event hub namespace</summary>
+        private static readonly Regex namespaceNameRegex = new Regex(@"sb://(?<namespacename>.+)\.servicebus\.windows\.net", RegexOptions.IgnoreCase | RegexOptions.Compiled);
+
+        /// <summary>
+        ///   It parses a connection string looking for the provided token name and returns its value.
+        /// </summary>
+        ///
+        /// <param name="connectionString">An Event Hubs namespace connection string</param>
+        /// <param name="tokenKey">The token to be looked for</param>
+        ///
+        /// <returns>The value of the token if found; otherwise, a <see cref="InvalidOperationException" /> is thrown.</returns>
+        ///
+        internal static string ParseTokenAndReturnValue(string connectionString, string tokenKey)
```

Comment:
(extreme) nit:  The class is `internal`,  we normally make the methods `public` since they'll be constrained by the class.  (that way, if we change visibility, the methods follow the class with no additional work)